pipit/readers/otf2_reader.py

```diff
@@ -230,15 +235,17 @@ def events_reader(self, rank_size):
 
             trace.close()  # close event files
 
-        # returns dictionary with all events and their fields
-        return {
-            "Timestamp (ns)": timestamps,
-            "Event Type": event_types,
-            "Name": names,
-            "Thread": thread_ids,
-            "Process": process_ids,
-            "Attributes": event_attributes,
-        }
+        # returns dataframe with all events and their fields
+        return pd.DataFrame(
```

Comment:
This change is being made because:
Following what Alex does, this removes the need for the previous combining of dictionaries, which was done confusingly and was ambiguous. It also has a slight edge in terms of performance.